Android/app/src/main/java/app/intra/util/DnsQueryTracker.java

```diff
@@ -60,10 +62,8 @@ public long getNumRequests() {
     }
   }
 
-  public Queue<Long> getRecentActivity() {
-    synchronized (this) {
-      return new LinkedList<>(recentActivity);
-    }
+  public synchronized void readInto(DnsActivityReader reader) {
```

Comment:
it seems like a bad idea to be using two different locking mechanisms here.  E.g. this isn't synchronizing on `this` while other methods are, and so recentActivity could be modified while this is running.

I'm fairly certain there's no risk of locking though.

Reply:
Marking a method as `synchronized` is equivalent to locking `this`, so it's not two different mechanisms.

However, since it's evidently confusing, I've done a cleanup pass and converted all the methods.

Reply:
Ah, I looked at the java documentation to try to find out if that was the case and didn't see it anywhere.